packages/@aws-cdk/aws-sns/lib/policy.ts

```diff
@@ -12,6 +12,11 @@ export interface TopicPolicyProps {
    * The set of topics this policy applies to.
    */
   readonly topics: ITopic[];
+  /**
+   * IAM policy document to apply to topic(s).
+   */
+  readonly policyDocument: PolicyDocument;
```

Comment:
Since the SNS module is stable, can we make this property optional? I know an empty policy isn't super useful but we cloud potentially consider emitting a warning if the user creates a policy with no document.

Something like [ASG is doing here](https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/aws-autoscaling/lib/auto-scaling-group.ts#L663)

Reply:
Actually, if synth/deploy is failing when policies are empty. The require property may be correct. The issue says it fails during synth. Is that correct? If so I think this makes sense.

Reply:
I did not check any synth errors, but I will check it and then report back here. Thanks

Reply:
Got the following error when creating instance of `TopicPolicy` without `policyDocument` prop. 

```
error: Resolution error: Supplied properties not correct for "CfnTopicPolicyProps"
  policyDocument: required but missing.
```

Reply:
@MrArnoldPalmer  Hello,  I tested and making it a required prop is the only way to go since omitted it results in synth error. thanks.